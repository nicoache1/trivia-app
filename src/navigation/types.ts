import { ParamListBase } from '@react-navigation/native'
import { NativeStackScreenProps } from '@react-navigation/native-stack'
import React from 'react'

import { Routes } from './routes'

export interface ParamList extends ParamListBase {
  [Routes.Welcome]: undefined
  [Routes.ConfigureGame]: undefined
  [Routes.Game]: undefined
  [Routes.Results]: undefined
  [Routes.AnimatedSplash]: undefined
}

export type SceneProps<T extends keyof ParamList> = React.FC<
  NativeStackScreenProps<ParamList, T>
>

export type NavigationProp<T extends keyof ParamList> = NativeStackScreenProps<
  ParamList,
  T
>['navigation']
