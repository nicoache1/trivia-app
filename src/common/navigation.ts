import {
  NavigationContainerRef,
  ParamListBase,
  StackActions,
} from '@react-navigation/native'
import * as React from 'react'

import { ParamList } from '../navigation/types'

export const navigationRef: React.RefObject<NavigationContainerRef<ParamList>> =
  React.createRef()

export function navigate(name: string, params?: ParamListBase[string]) {
  navigationRef.current?.navigate(name, params)
}

export function push(name: string, params?: ParamListBase[string]) {
  navigationRef.current?.dispatch(StackActions.push(name, params))
}

export function popToTop() {
  navigationRef.current?.dispatch(StackActions.popToTop())
}

export function pop() {
  navigationRef.current?.dispatch(StackActions.pop())
}
