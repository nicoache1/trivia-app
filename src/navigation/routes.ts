export enum Routes {
  Game = 'Game',
  ConfigureGame = 'ConfigureGame',
  AnimatedSplash = 'AnimatedSplash',
  Results = 'Results',
  Welcome = 'Welcome',
}
