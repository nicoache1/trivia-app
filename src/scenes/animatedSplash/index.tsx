import SplashImage from 'assets/splashShape.png'
import * as SplashScreen from 'expo-splash-screen'
import React, { useLayoutEffect } from 'react'
import { Dimensions, StyleSheet, View } from 'react-native'
import Animated, {
  interpolate,
  runOnJS,
  useAnimatedStyle,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated'
import { Routes } from 'src/navigation/routes'
import { SceneProps } from 'src/navigation/types'
import { useTheme } from 'src/styles/Theme'

import { styles } from './styles'

const { width, height } = Dimensions.get('window')

export const AnimatedSplash: SceneProps<Routes.AnimatedSplash> = ({
  navigation,
}) => {
  useLayoutEffect(() => {
    navigation.setOptions({ headerShown: false })
  }, [navigation])

  const {
    Theme: { colors },
  } = useTheme()

  const progress = useSharedValue(0)
  const imageAnimation = useSharedValue(0)

  const onImageLoaded = async () => {
    await SplashScreen.hideAsync()
    progress.value = withTiming(1, { duration: 700 }, () => {
      imageAnimation.value = withTiming(1, {}, finished => {
        if (finished) {
          runOnJS(navigation.replace)(Routes.Welcome)
        }
      })
    })
  }

  const animatedStyle = useAnimatedStyle(() => ({
    backgroundColor: colors.SECONDARY,
    transform: [
      { translateY: width / 2 },
      { rotate: `${interpolate(progress.value, [0, 1], [-25, 0])}deg` },
      { translateX: interpolate(progress.value, [0, 1], [-height, 0]) },
      { translateY: -width / 2 },
    ],
    zIndex: 1,
  }))

  const animatedImage = useAnimatedStyle(() => ({
    height: interpolate(imageAnimation.value, [0, 0.5, 1], [70, 80, 0]),
    opacity: interpolate(imageAnimation.value, [0, 1], [1, 0]),
    zIndex: 100,
  }))

  return (
    <View style={styles.container}>
      <Animated.Image
        onLoadEnd={onImageLoaded}
        source={SplashImage}
        style={animatedImage}
        resizeMode={'contain'}
      />
      <Animated.View
        style={[animatedStyle, { ...StyleSheet.absoluteFillObject }]}
      />
    </View>
  )
}
