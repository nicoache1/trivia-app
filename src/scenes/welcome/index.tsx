import Circle from 'assets/onboarding/circle.svg'
import { localization } from 'localization'
import React, { useLayoutEffect, useState } from 'react'
import { Dimensions, View } from 'react-native'
import Animated, {
  interpolateColor,
  useAnimatedStyle,
  useDerivedValue,
  useSharedValue,
  withTiming,
} from 'react-native-reanimated'
import { StyledButton } from 'src/components/StyledButton'
import { StyledContainer } from 'src/components/StyledContainer'
import { Routes } from 'src/navigation/routes'
import { SceneProps } from 'src/navigation/types'
import { Palette } from 'src/styles/Palette'

import { Slide1 } from './Slide1'
import { Slide2 } from './Slide2'
import { Slide3 } from './Slide3'
import { Slide4 } from './Slide4'
import { styles } from './styles'

const { width, height } = Dimensions.get('window')

const colors = [
  Palette.SECONDARY,
  Palette.ORANGE,
  Palette.PRIMARY,
  Palette.PINK,
]
const buttonColors = [
  Palette.PRIMARY,
  Palette.SECONDARY,
  Palette.ORANGE,
  Palette.PRIMARY,
]
const snapPoints = [0, -width, -width * 2, -width * 3]
const circleYPositions = [height / 5, height / 3, height / 5, height / 6]
const circleXPositions = [-20, -40, 10, -30]
const circleScaleSizes = [1, 1.5, 1.2, 1.4]
const rotations = [-40, 0, 0, -40]
const buttonText = [
  localization('startNowLabel'),
  localization('nextLabel'),
  localization('nextLabel'),
  localization('beginLabel'),
]

export const Welcome: SceneProps<Routes.Welcome> = ({ navigation }) => {
  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: false,
    })
  }, [navigation])

  const translateX = useSharedValue(0)
  const circleTranslateY = useSharedValue(height / 5)
  const circleTranslateX = useSharedValue(-20)
  const circleScale = useSharedValue(1)
  const circleRotate = useSharedValue(-40)
  const [index, setIndex] = useState<number>(0)

  const derivedBackground = useDerivedValue(() => {
    const WIDTH_3 = width * 3
    const WIDTH_2 = width * 2
    const WIDTH_1 = width

    return interpolateColor(
      translateX.value * -1,
      [0, WIDTH_1, WIDTH_2, WIDTH_3],
      colors,
      'RGB',
    )
  })

  const derivedButtonColor = useDerivedValue(() => {
    const WIDTH_3 = width * 3
    const WIDTH_2 = width * 2
    const WIDTH_1 = width

    return interpolateColor(
      translateX.value * -1,
      [0, WIDTH_1, WIDTH_2, WIDTH_3],
      buttonColors,
      'RGB',
    )
  })

  const animatedBackgroundStyle = useAnimatedStyle(
    () => ({
      backgroundColor: derivedBackground.value,
    }),
    [derivedButtonColor],
  )

  const animatedButtonStyle = useAnimatedStyle(
    () => ({
      backgroundColor: derivedButtonColor.value,
    }),
    [derivedButtonColor],
  )

  const onPress = () => {
    if (index !== 3) {
      const newIndex = index + 1

      const value = snapPoints[newIndex]
      const circleYPosition = circleYPositions[newIndex]
      const circleXPosition = circleXPositions[newIndex]
      const circleScaleSize = circleScaleSizes[newIndex]
      const circleRotation = rotations[newIndex]

      translateX.value = withTiming(value, { duration: 400 })

      circleTranslateY.value = withTiming(circleYPosition, { duration: 500 })
      circleTranslateX.value = withTiming(circleXPosition, { duration: 500 })
      circleScale.value = withTiming(circleScaleSize, { duration: 500 })
      circleRotate.value = withTiming(circleRotation, { duration: 500 })

      setIndex(newIndex)
    } else {
      navigation.navigate(Routes.ConfigureGame)
    }
  }

  const animatedStyle = useAnimatedStyle(() => ({
    transform: [{ translateX: translateX.value }],
  }))

  const circleAnimatedStyle = useAnimatedStyle(() => ({
    right: circleTranslateX.value,
    top: circleTranslateY.value,
    transform: [
      { scale: circleScale.value },
      { rotate: `${circleRotate.value}deg` },
    ],
  }))

  return (
    <Animated.View style={[styles.container, animatedBackgroundStyle]}>
      <View style={styles.header} />
      <Animated.View style={[styles.circle, circleAnimatedStyle]}>
        <Circle />
      </Animated.View>
      <View style={styles.slider}>
        <Animated.View style={[styles.sliderContainer, animatedStyle]}>
          <Slide1 />
          <Slide2 />
          <Slide3 />
          <Slide4 />
        </Animated.View>
      </View>
      <StyledContainer style={styles.footer}>
        <StyledButton
          onPress={onPress}
          customStyles={animatedButtonStyle}
          label={buttonText[index]}
        />
      </StyledContainer>
    </Animated.View>
  )
}
