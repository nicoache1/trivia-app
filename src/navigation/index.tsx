import { createNativeStackNavigator } from '@react-navigation/native-stack'
import React from 'react'

import { AnimatedSplash } from '../scenes/animatedSplash'
import { ConfigureGame } from '../scenes/configureGame'
import { Game } from '../scenes/game'
import { Results } from '../scenes/results'
import { Welcome } from '../scenes/welcome'
import { Routes } from './routes'
import { ParamList } from './types'

const Stack = createNativeStackNavigator<ParamList>()

export const AppNavigator: React.FC<{}> = ({}) => {
  return (
    <Stack.Navigator
      screenOptions={{
        contentStyle: {
          backgroundColor: 'white',
        },
      }}>
      <Stack.Screen name={Routes.AnimatedSplash} component={AnimatedSplash} />
      <Stack.Screen
        name={Routes.Welcome}
        component={Welcome}
        options={{ presentation: 'modal' }}
      />
      <Stack.Screen name={Routes.ConfigureGame} component={ConfigureGame} />
      <Stack.Screen name={Routes.Game} component={Game} />
      <Stack.Screen name={Routes.Results} component={Results} />
    </Stack.Navigator>
  )
}
