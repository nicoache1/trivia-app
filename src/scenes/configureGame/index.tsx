import ConfigureImage from 'assets/configure.png'
import { localization } from 'localization'
import React, { useLayoutEffect } from 'react'
import { Controller, useForm } from 'react-hook-form'
import { Dimensions, Image, View } from 'react-native'
import { Background } from 'src/components/Background'
import { SceneContainer } from 'src/components/SceneContainer'
import { StyledButton } from 'src/components/StyledButton'
import { ButtonMode } from 'src/components/StyledButton/ButtonMode'
import { StyledContainer } from 'src/components/StyledContainer'
import { StyledText } from 'src/components/StyledText'
import { Routes } from 'src/navigation/routes'
import { SceneProps } from 'src/navigation/types'
import { useAppDispatch } from 'src/store'
import { setGameAction } from 'src/store/storage/game'
import { useTheme } from 'src/styles/Theme'
import { AppTypography, ColorPalette } from 'src/styles/types'
import { Game, getDifficulties, getQuestionAmounts } from 'src/types/game'

import { renderInputControl, renderSegmentedControl } from './formRenders'
import { styles } from './styles'
import { ConfigureGameForm } from './types'

const { width, height } = Dimensions.get('window')

export const ConfigureGame: SceneProps<Routes.ConfigureGame> = ({
  navigation,
}) => {
  const {
    Theme: { colors },
  } = useTheme()

  const dispatch = useAppDispatch()

  useLayoutEffect(() => {
    navigation.setOptions({
      headerShown: false,
    })
  }, [navigation])

  const { control, handleSubmit } = useForm<ConfigureGameForm>({
    defaultValues: {
      amount: 0,
      difficulty: 0,
      name: localization('defaultName'),
    },
  })

  const difficulties = getDifficulties()
  const questions = getQuestionAmounts()

  const onSubmit = (data: ConfigureGameForm) => {
    const newGame: Game = {
      difficulty: difficulties[data.difficulty],
      player: data.name,
      questionAmount: questions[data.amount],
      results: [],
    }
    dispatch(setGameAction(newGame))
    navigation.navigate(Routes.Game)
  }

  return (
    <View
      style={[styles.screenContainer, { backgroundColor: colors.SECONDARY }]}>
      <SceneContainer style={styles.scene} barStyle={'light-content'}>
        <View style={styles.headerContainer}>
          <Background />
          <Image
            source={ConfigureImage}
            style={{ height: height / 2.5, width: width * 0.9 }}
            resizeMode={'contain'}
          />
          <View style={styles.textContainer}>
            <StyledText
              color={ColorPalette.TERTIARY}
              typography={AppTypography.H1}>
              {localization('configureTitle1')}
            </StyledText>
            <StyledText
              color={ColorPalette.WHITE}
              typography={AppTypography.H1}>
              {localization('configureTitle2')}
            </StyledText>
          </View>
        </View>
        <View style={styles.formContainer}>
          <Controller
            control={control}
            rules={{
              required: true,
            }}
            render={renderInputControl}
            name={'name'}
          />
          <StyledContainer>
            <View style={styles.subtitle}>
              <StyledText
                color={ColorPalette.WHITE}
                typography={AppTypography.P1_BOLD}>
                {localization('difficultyLabel')}
              </StyledText>
            </View>
            <Controller
              control={control}
              rules={{
                required: true,
              }}
              render={renderSegmentedControl<'difficulty'>(
                difficulties,
                colors.WHITE,
              )}
              name={'difficulty'}
            />
          </StyledContainer>
          <StyledContainer>
            <View style={styles.subtitle}>
              <StyledText
                color={ColorPalette.WHITE}
                typography={AppTypography.P1_BOLD}>
                {localization('questionAmountLabel')}
              </StyledText>
            </View>
            <Controller
              control={control}
              rules={{
                required: true,
              }}
              render={renderSegmentedControl<'amount'>(
                questions.map(item => `${item}`),
                colors.WHITE,
              )}
              name={'amount'}
            />
          </StyledContainer>
        </View>
        <StyledContainer style={styles.buttonContainer}>
          <StyledButton
            mode={ButtonMode.DARK}
            label={localization('getStartedLabel')}
            onPress={handleSubmit(onSubmit)}
          />
        </StyledContainer>
      </SceneContainer>
    </View>
  )
}
