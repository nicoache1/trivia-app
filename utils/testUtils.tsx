import type { PreloadedState } from '@reduxjs/toolkit'
import type { RenderOptions } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import { AppStore, RootState, setupStore } from 'src/store'

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export function renderWithProvider(
  ui: React.ReactElement,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
  }: ExtendedRenderOptions = {},
) {
  return <Provider store={store}>{ui}</Provider>
}
