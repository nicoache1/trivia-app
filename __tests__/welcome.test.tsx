import { render } from '@testing-library/react-native'
import React from 'react'
import { Routes } from 'src/navigation/routes'
import { Welcome } from 'src/scenes/welcome'

import {
  createNavigationPropMock,
  createRoutePropMock,
} from '../__mocks__/navigationMocks'
import { renderWithProvider } from '../utils/testUtils'

test('Expect Welcome to be rendered correctly with default', async () => {
  const mockRoute = createRoutePropMock<Routes.Welcome>(Routes.Welcome)

  const mockNavigation = createNavigationPropMock<Routes.Welcome>()

  const enhancedComponent = renderWithProvider(
    <Welcome route={mockRoute} navigation={mockNavigation} />,
  )

  const renderedSnapshot = render(enhancedComponent).toJSON()

  expect(renderedSnapshot).toMatchSnapshot()
})
