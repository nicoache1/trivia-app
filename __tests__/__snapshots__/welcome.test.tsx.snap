// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect Welcome to be rendered correctly with default 1`] = `
<View
  animatedStyle={
    Object {
      "value": Object {
        "backgroundColor": "rgba(108, 92, 231, 1)",
      },
    }
  }
  collapsable={false}
  style={
    Object {
      "backgroundColor": "rgba(108, 92, 231, 1)",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 0.2,
      }
    }
  />
  <View
    animatedStyle={
      Object {
        "value": Object {
          "right": -20,
          "top": 266.8,
          "transform": Array [
            Object {
              "scale": 1,
            },
            Object {
              "rotate": "-40deg",
            },
          ],
        },
      }
    }
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "position": "absolute",
        "right": -20,
        "top": 266.8,
        "transform": Array [
          Object {
            "scale": 1,
          },
          Object {
            "rotate": "-40deg",
          },
        ],
      }
    }
  >
    <SvgMock />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "height": 1334,
      }
    }
  >
    <View
      animatedStyle={
        Object {
          "value": Object {
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          },
        }
      }
      collapsable={false}
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="717"
            bbWidth={750}
            fill="none"
            focusable={false}
            height="717"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 717,
                  "width": 750,
                },
              ]
            }
            vbHeight={717}
            vbWidth={375}
            width={750}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M-348.5 688.5L-801 319.5V1024.5H868V340.5L521 749L73.5 0L-348.5 688.5Z"
                fill={4279510628}
                fillOpacity={0.2}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <SvgMock />
        <Text
          accessible={true}
          color="TERTIARY"
          style={
            Object {
              "color": "#142664",
              "fontSize": 62,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H0"
        >
          welcomeLabel
        </Text>
        <Text
          accessible={true}
          color="WHITE"
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "300",
            }
          }
          typography="H1"
        >
          tricktyLabel
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={1334}
            bbWidth={750}
            fill="none"
            focusable={false}
            height={1334}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 1334,
                  "width": 750,
                },
              ]
            }
            vbHeight={812}
            vbWidth={375}
            width={750}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M-7.5 612.5L-460 243.5V948.5H1209V264.5L862 673L414.5 -76L-7.5 612.5Z"
                fill={4293161580}
                fillOpacity={0.4}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <SvgMock />
        <Text
          accessible={true}
          color="WHITE"
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H1"
        >
          comeOnLabel
        </Text>
        <Text
          accessible={true}
          color="TERTIARY"
          style={
            Object {
              "color": "#142664",
              "fontSize": 62,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H0"
        >
          itsEasyLabel
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight={1334}
            bbWidth={750}
            fill="none"
            focusable={false}
            height={1334}
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 1334,
                  "width": 750,
                },
              ]
            }
            vbHeight={812}
            vbWidth={375}
            width={750}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M-382.5 612.5L-835 243.5V948.5H834V264.5L487 673L39.5 -76L-382.5 612.5Z"
                fill={4278231425}
                fillOpacity={0.4}
                propList={
                  Array [
                    "fill",
                    "fillOpacity",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <SvgMock />
        <Text
          accessible={true}
          color="TERTIARY"
          style={
            Object {
              "color": "#142664",
              "fontSize": 62,
              "fontStyle": "normal",
              "fontWeight": "700",
            }
          }
          typography="H0"
        >
          playLabel
        </Text>
        <Text
          accessible={true}
          color="WHITE"
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 32,
              "fontStyle": "normal",
              "fontWeight": "300",
              "textAlign": "center",
            }
          }
          typography="H1"
        >
          itsFunnyLabel
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
            "width": 750,
          }
        }
      >
        <View
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
            }
          }
        >
          <RNSVGSvgView
            align="xMidYMid"
            bbHeight="812"
            bbWidth={750}
            fill="none"
            focusable={false}
            height="812"
            meetOrSlice={0}
            minX={0}
            minY={0}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                Object {
                  "flex": 0,
                  "height": 812,
                  "width": 750,
                },
              ]
            }
            vbHeight={812}
            vbWidth={375}
            width={750}
          >
            <RNSVGGroup
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            >
              <RNSVGPath
                d="M-757.5 612.5L-1210 243.5V948.5H459V264.5L112 673L-335.5 -76L-757.5 612.5Z"
                fill={4293287565}
                propList={
                  Array [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <SvgMock />
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            color="TERTIARY"
            style={
              Object {
                "color": "#142664",
                "fontSize": 32,
                "fontStyle": "normal",
                "fontWeight": "700",
              }
            }
            typography="H1"
          >
            letsLabel
          </Text>
          <Text
            accessible={true}
            color="WHITE"
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 32,
                "fontStyle": "normal",
                "fontWeight": "700",
              }
            }
            typography="H1"
          >
            startLabel
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "bottom": 53.36,
        "height": 133.4,
        "justifyContent": "flex-start",
        "left": 0,
        "paddingHorizontal": 20,
        "paddingVertical": 10,
        "position": "absolute",
        "right": 0,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 56,
            "justifyContent": "center",
            "paddingVertical": 10,
            "width": "100%",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    >
      <View
        animatedStyle={
          Object {
            "value": Object {
              "backgroundColor": "rgba(0, 184, 148, 1)",
            },
          }
        }
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 184, 148, 1)",
            "borderRadius": 10,
            "height": 56,
            "justifyContent": "center",
            "paddingVertical": 10,
            "width": "100%",
          }
        }
      >
        <Text
          accessible={true}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontSize": 16,
                "fontStyle": "normal",
                "fontWeight": "700",
              },
              Object {
                "alignItems": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
          typography="P1_BOLD"
        >
          startNowLabel
        </Text>
      </View>
    </View>
  </View>
</View>
`;
